### PR TITLE
Fix sending long messages - TS edition

### PR DESCRIPTION
**Pull request by Half-Shot**
Supersedes #30

This basically is an implementation of ^ but written for the TS world.

Thanks to @xzfc for the initial PR.

### CODE REVIEW

File: src/splitLines.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+function getUtf8LenAt(text: string, cutPosCurr: number) {
```
Comment (jaller94) [inline src/splitLines.ts:1]: If this is meant to be complete: 4byte UTF8 chars do exist and are common. 👋

File: src/splitLines.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const result = [];
+
+    let cutPosStart = 0, bytesStart = 0;
+    let cutPosWord = 0, bytesWord = 0;
+    let cutPosCurr = 0, bytesCurr = 0;
+    let wasSpace = true;
+
+    while (cutPosCurr < text.length) {
+        let utf8len, utf16len, isWhitespace;
+        if ((text.charCodeAt(cutPosCurr) & 0xF800) === 0xD800) {
```
Comment (jaller94) [inline src/splitLines.ts:30]: I guess this covers 4byte utf8 characters.

File: src/splitLines.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            bytesCurr += utf8len;
+            wasSpace = isWhitespace;
+        }
+    }
+
+
+    if (cutPosStart !== cutPosCurr) {
+        let startPos = cutPosStart;
+        if (result.length !== 0) {
+            while (startPos < text.length && text[startPos].match(/\s/)) {
```
Comment (jaller94) [inline src/splitLines.ts:81]:
```suggestion
            while (startPos < text.length && /\s/.test(text[startPos])) {
```

**Review by jaller94 — APPROVED** [review]
Looks good. Nice tests! :tada:

File: src/splitLines.ts (hunk 2 above)
Comment (Half-Shot) [reply src/splitLines.ts:30]: Yup, it's all a bit gnarly on IRC.

**Comment by Half-Shot** [comment]
I broke lint on master, will fix in subsequent PR.

**Pull request merged**